### PR TITLE
Added setting for scroll acceleration

### PR DESCRIPTION
**Pull request by AndreasArvidsson**

### CODE REVIEW

File: plugin/mouse/mouse_scroll.py [hunk 1]
```diff
@@ -184,7 +195,13 @@ def mouse_scroll_continuous(new_scroll_dir: Literal[-1, 1]):
...
 def scroll_continuous_helper():
     scroll_amount = settings.get("user.mouse_continuous_scroll_amount")
-    y = scroll_amount * scroll_dir / 10
+    acceleration_setting = settings.get("user.mouse_continuous_scroll_acceleration")
+    acceleration_speed = (
+        1 + min((time.perf_counter() - scroll_ts) / 0.5, acceleration_setting - 1)
+        if acceleration_setting > 1
+        else 1
+    )
+    y = scroll_amount * acceleration_speed * scroll_dir / 10
```
Comment (phillco) [inline plugin/mouse/mouse_scroll.py:204]: I know the existing code does this, but do we know why we have a setting for `mouse_continuous_scroll_amount` if we're just going to divide it by 10 arbitrarily 😭. Maybe the setting used to be different and we needed to change its meaning?

File: settings.talon [hunk 2]
```diff
@@ -24,6 +24,9 @@ settings():
     # Set the scroll amount for continuous scroll/gaze scroll
     user.mouse_continuous_scroll_amount = 80
 
+    # Set the maximum acceleration factor when scrolling continuously. 1=constant speed/no acceleration.
+    user.mouse_continuous_scroll_acceleration = 1
```
Comment (phillco) [inline settings.talon:28]: Could we document the type of acceleration (ie linear)?

File: plugin/mouse/mouse_scroll.py [hunk 3]
```diff
@@ -7,6 +8,7 @@
 scroll_job = None
 gaze_job = None
 scroll_dir: Literal[-1, 1] = 1
+scroll_ts: float = 0
```
Comment (phillco) [inline plugin/mouse/mouse_scroll.py:11]: Thought about suggesting `scroll_start_ts` which is a tiny bit clearer but uglier. Not strongly opinionated. Feel free to go either way.

File: plugin/mouse/mouse_scroll.py (hunk 1 above)
Comment (AndreasArvidsson) [reply plugin/mouse/mouse_scroll.py:204]: I have no idea why and was very tempted to change it, but that would of course break every one who has changed the default setting.

File: settings.talon (hunk 2 above)
Comment (AndreasArvidsson) [reply settings.talon:28]: done

File: plugin/mouse/mouse_scroll.py (hunk 3 above)
Comment (AndreasArvidsson) [reply plugin/mouse/mouse_scroll.py]: done

File: plugin/mouse/mouse_scroll.py (hunk 1 above)
Comment (knausj85) [reply plugin/mouse/mouse_scroll.py:204]: One of many great mysteries lost to time. Might be worth writing up an issue if either if you feel strongly that we should revisit

**Review by knausj85 — APPROVED** [review]
It works as advertised.

File: plugin/mouse/mouse_scroll.py (hunk 1 above)
Comment (AndreasArvidsson) [reply plugin/mouse/mouse_scroll.py:204]: Agreed
https://github.com/talonhub/community/issues/1605

**Pull request merged**